### PR TITLE
Full implementation of ParseUIDemo in Swift.

### PR DESCRIPTION
**Pull request by nlutsenko-fb**
This implements ParseUIDemo in Swift alongside the existing Obj-C demo.
cc @grantland @stanleyw @hallucinogen

### CODE REVIEW

File: ParseUIDemo/Swift/CustomViewControllers/ProductTableViewController/CustomProductTableViewController.swift [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import Parse
+import ParseUI
+
+class CustomProductTableViewController: PFProductTableViewController {
+
+    override func tableView(tableView: UITableView, didSelectRowAtIndexPath indexPath: NSIndexPath) {
+        let product = objects?[indexPath.row] as? PFObject
+        if let identifier = product?["productIdentifier"] as? String {
+            if identifier == "Cooper" {
+                PFPurchase.buyProduct(identifier) { error in
```
Comment (hallucinogen) [inline ParseUIDemo/Swift/CustomViewControllers/ProductTableViewController/CustomProductTableViewController.swift]: ? What does this mean?

File: ParseUIDemo/Swift/UIDemoViewController.swift [hunk 2]
```diff
@@ -0,0 +1,352 @@
...
+        return 1
+    }
+
+    override func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return UIDemoType.count
+    }
+
+    override func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCellWithIdentifier("cell") as! UITableViewCell
+        cell.textLabel?.text = UIDemoType(rawValue: indexPath.row)?.description
```
Comment (hallucinogen) [inline ParseUIDemo/Swift/UIDemoViewController.swift]: cell?

File: ParseUIDemo/Swift/CustomViewControllers/QueryCollectionViewController/PaginatedCollectionViewController.swift [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+    override func queryForCollection() -> PFQuery {
+        return super.queryForCollection().orderByAscending("priority")
+    }
+
+    // MARK: CollectionView
+
+    override func collectionView(collectionView: UICollectionView, cellForItemAtIndexPath indexPath: NSIndexPath, object: PFObject?) -> PFCollectionViewCell? {
+        let cell = super.collectionView(collectionView, cellForItemAtIndexPath: indexPath, object: object)
+        cell?.textLabel.textAlignment = .Center
```
Comment (hallucinogen) [inline ParseUIDemo/Swift/CustomViewControllers/QueryCollectionViewController/PaginatedCollectionViewController.swift]: `textLabel?` ?

File: ParseUIDemo/Swift/CustomViewControllers/QueryCollectionViewController/PaginatedCollectionViewController.swift [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+        if let title = object?["title"] as? String {
+            let attributedTitle = NSMutableAttributedString(string: title)
+            if var priority = object?["priority"] as? Int {
+                let attributes = [NSFontAttributeName : UIFont.systemFontOfSize(13.0), NSForegroundColorAttributeName : UIColor.grayColor()]
+                let string = NSAttributedString(string: "\nPriority: \(priority)", attributes: attributes)
+                attributedTitle.appendAttributedString(string)
+            }
+            cell?.textLabel.attributedText = attributedTitle
```
Comment (hallucinogen) [inline ParseUIDemo/Swift/CustomViewControllers/QueryCollectionViewController/PaginatedCollectionViewController.swift]: `textLabel?` ?

File: ParseUIDemo/Swift/CustomViewControllers/QueryCollectionViewController/PaginatedCollectionViewController.swift [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+        } else {
+            cell?.textLabel.attributedText = NSAttributedString()
```
Comment (hallucinogen) [inline ParseUIDemo/Swift/CustomViewControllers/QueryCollectionViewController/PaginatedCollectionViewController.swift]: `textLabel?` ?

File: ParseUIDemo/Swift/CustomViewControllers/QueryCollectionViewController/SectionedCollectionViewController.swift [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+
+    override func collectionView(collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        let array = sections[sectionKeys[section]]
+        return array?.count ?? 0
+    }
+
+    override func collectionView(collectionView: UICollectionView, cellForItemAtIndexPath indexPath: NSIndexPath, object: PFObject?) -> PFCollectionViewCell? {
+        let cell = super.collectionView(collectionView, cellForItemAtIndexPath: indexPath, object: object)
+
+        cell?.textLabel.textAlignment = .Center
```
Comment (hallucinogen) [inline ParseUIDemo/Swift/CustomViewControllers/QueryCollectionViewController/SectionedCollectionViewController.swift]: `textLabel?` ?

File: ParseUIDemo/Swift/CustomViewControllers/QueryCollectionViewController/SectionedCollectionViewController.swift [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+        cell?.textLabel.text = object?["title"] as? String
```
Comment (hallucinogen) [inline ParseUIDemo/Swift/CustomViewControllers/QueryCollectionViewController/SectionedCollectionViewController.swift]: `textLabel?` ?

File: ParseUIDemo/Swift/CustomViewControllers/QueryCollectionViewController/SimpleCollectionViewController.swift [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+
+    override func queryForCollection() -> PFQuery {
+        return super.queryForCollection().orderByAscending("priority")
+    }
+
+    // MARK: CollectionView
+
+    override func collectionView(collectionView: UICollectionView, cellForItemAtIndexPath indexPath: NSIndexPath, object: PFObject?) -> PFCollectionViewCell? {
+        let cell = super.collectionView(collectionView, cellForItemAtIndexPath: indexPath, object: object)
+        cell?.textLabel.textAlignment = .Center
```
Comment (hallucinogen) [inline ParseUIDemo/Swift/CustomViewControllers/QueryCollectionViewController/SimpleCollectionViewController.swift]: `textLabel?` ?

File: ParseUIDemo/Swift/CustomViewControllers/QueryCollectionViewController/SimpleCollectionViewController.swift [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+
+        if let title = object?["title"] as? String {
+            let attributedTitle = NSMutableAttributedString(string: title)
+            if var priority = object?["priority"] as? Int {
+                let attributes = [NSFontAttributeName : UIFont.systemFontOfSize(13.0), NSForegroundColorAttributeName : UIColor.grayColor()]
+                let string = NSAttributedString(string: "\nPriority: \(priority)", attributes: attributes)
+                attributedTitle.appendAttributedString(string)
+            }
+            cell?.textLabel.attributedText = attributedTitle
```
Comment (hallucinogen) [inline ParseUIDemo/Swift/CustomViewControllers/QueryCollectionViewController/SimpleCollectionViewController.swift]: `textLabel?` ?

File: ParseUIDemo/Swift/CustomViewControllers/QueryCollectionViewController/SimpleCollectionViewController.swift [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+        } else {
+            cell?.textLabel.attributedText = NSAttributedString()
```
Comment (hallucinogen) [inline ParseUIDemo/Swift/CustomViewControllers/QueryCollectionViewController/SimpleCollectionViewController.swift]: `textLabel?` ?

File: ParseUIDemo/Swift/CustomViewControllers/QueryCollectionViewController/StoryboardCollectionViewController.swift [hunk 11]
```diff
@@ -0,0 +1,77 @@
...
+
+    override func queryForCollection() -> PFQuery {
+        return super.queryForCollection().orderByAscending("priority")
+    }
+
+    // MARK: CollectionView
+
+    override func collectionView(collectionView: UICollectionView, cellForItemAtIndexPath indexPath: NSIndexPath, object: PFObject?) -> PFCollectionViewCell? {
+        let cell = super.collectionView(collectionView, cellForItemAtIndexPath: indexPath, object: object)
+        cell?.textLabel.textAlignment = .Center
```
Comment (hallucinogen) [inline ParseUIDemo/Swift/CustomViewControllers/QueryCollectionViewController/StoryboardCollectionViewController.swift]: `textLabel?` ?

File: ParseUIDemo/Swift/CustomViewControllers/QueryCollectionViewController/StoryboardCollectionViewController.swift [hunk 12]
```diff
@@ -0,0 +1,77 @@
...
+
+        if let title = object?["title"] as? String {
+            let attributedTitle = NSMutableAttributedString(string: title)
+            if var priority = object?["priority"] as? Int {
+                let attributes = [NSFontAttributeName : UIFont.systemFontOfSize(13.0), NSForegroundColorAttributeName : UIColor.grayColor()]
+                let string = NSAttributedString(string: "\nPriority: \(priority)", attributes: attributes)
+                attributedTitle.appendAttributedString(string)
+            }
+            cell?.textLabel.attributedText = attributedTitle
```
Comment (hallucinogen) [inline ParseUIDemo/Swift/CustomViewControllers/QueryCollectionViewController/StoryboardCollectionViewController.swift]: `textLabel?` ?

File: ParseUIDemo/Swift/CustomViewControllers/QueryCollectionViewController/StoryboardCollectionViewController.swift [hunk 13]
```diff
@@ -0,0 +1,77 @@
...
+        } else {
+            cell?.textLabel.attributedText = NSAttributedString()
```
Comment (hallucinogen) [inline ParseUIDemo/Swift/CustomViewControllers/QueryCollectionViewController/StoryboardCollectionViewController.swift]: `textLabel?` ?

File: ParseUIDemo/Swift/CustomViewControllers/QueryCollectionViewController/SubtitleImageCollectionViewController.swift [hunk 14]
```diff
@@ -0,0 +1,73 @@
...
+        let sideLength = min(CGRectGetWidth(bounds), CGRectGetHeight(bounds)) / 2.0 - layout.minimumInteritemSpacing
+        layout.itemSize = CGSizeMake(sideLength, sideLength)
+    }
+
+    // MARK: CollectionView
+
+    override func collectionView(collectionView: UICollectionView, cellForItemAtIndexPath indexPath: NSIndexPath, object: PFObject?) -> PFCollectionViewCell? {
+        let cell = super.collectionView(collectionView, cellForItemAtIndexPath: indexPath, object: object)
+
+        cell?.textLabel.textAlignment = .Center
```
Comment (hallucinogen) [inline ParseUIDemo/Swift/CustomViewControllers/QueryCollectionViewController/SubtitleImageCollectionViewController.swift]: `textLabel?` ?

File: ParseUIDemo/Swift/CustomViewControllers/QueryCollectionViewController/SubtitleImageCollectionViewController.swift [hunk 15]
```diff
@@ -0,0 +1,73 @@
...
+        cell?.textLabel.text = object?["name"] as? String
```
Comment (hallucinogen) [inline ParseUIDemo/Swift/CustomViewControllers/QueryCollectionViewController/SubtitleImageCollectionViewController.swift]: `textLabel?` ?

File: ParseUIDemo/Swift/CustomViewControllers/QueryCollectionViewController/SubtitleImageCollectionViewController.swift [hunk 16]
```diff
@@ -0,0 +1,73 @@
...
+
+        cell?.imageView.file = object?["icon"] as? PFFile
```
Comment (hallucinogen) [inline ParseUIDemo/Swift/CustomViewControllers/QueryCollectionViewController/SubtitleImageCollectionViewController.swift]: `imageView?` ?

File: ParseUIDemo/Swift/UIDemoViewController.swift (hunk 2 above)
Comment (nlutsenko-fb) [inline ParseUIDemo/Swift/UIDemoViewController.swift]: (y)

File: ParseUIDemo/Swift/CustomViewControllers/QueryCollectionViewController/PaginatedCollectionViewController.swift (hunk 3 above)
Comment (nlutsenko-fb) [inline ParseUIDemo/Swift/CustomViewControllers/QueryCollectionViewController/PaginatedCollectionViewController.swift]: `textLabel` here is an implicitly unwrapped optional - so no need for `?`

File: ParseUIDemo/Swift/CustomViewControllers/QueryCollectionViewController/PaginatedCollectionViewController.swift (hunk 4 above)
Comment (nlutsenko-fb) [inline ParseUIDemo/Swift/CustomViewControllers/QueryCollectionViewController/PaginatedCollectionViewController.swift]: `textLabel` here is an implicitly unwrapped optional - so no need for `?`

File: ParseUIDemo/Swift/CustomViewControllers/QueryCollectionViewController/PaginatedCollectionViewController.swift (hunk 5 above)
Comment (nlutsenko-fb) [inline ParseUIDemo/Swift/CustomViewControllers/QueryCollectionViewController/PaginatedCollectionViewController.swift]: `textLabel` here is an implicitly unwrapped optional - so no need for `?`

File: ParseUIDemo/Swift/CustomViewControllers/QueryCollectionViewController/SectionedCollectionViewController.swift (hunk 6 above)
Comment (nlutsenko-fb) [inline ParseUIDemo/Swift/CustomViewControllers/QueryCollectionViewController/SectionedCollectionViewController.swift]: `textLabel` here is an implicitly unwrapped optional - so no need for `?`

File: ParseUIDemo/Swift/CustomViewControllers/QueryCollectionViewController/SectionedCollectionViewController.swift (hunk 7 above)
Comment (nlutsenko-fb) [inline ParseUIDemo/Swift/CustomViewControllers/QueryCollectionViewController/SectionedCollectionViewController.swift]: `textLabel` here is an implicitly unwrapped optional - so no need for `?`

File: ParseUIDemo/Swift/CustomViewControllers/QueryCollectionViewController/SimpleCollectionViewController.swift (hunk 8 above)
Comment (nlutsenko-fb) [inline ParseUIDemo/Swift/CustomViewControllers/QueryCollectionViewController/SimpleCollectionViewController.swift]: `textLabel` here is an implicitly unwrapped optional - so no need for `?`

File: ParseUIDemo/Swift/CustomViewControllers/QueryCollectionViewController/SimpleCollectionViewController.swift (hunk 9 above)
Comment (nlutsenko-fb) [inline ParseUIDemo/Swift/CustomViewControllers/QueryCollectionViewController/SimpleCollectionViewController.swift]: `textLabel` here is an implicitly unwrapped optional - so no need for `?`

File: ParseUIDemo/Swift/CustomViewControllers/QueryCollectionViewController/SimpleCollectionViewController.swift (hunk 10 above)
Comment (nlutsenko-fb) [inline ParseUIDemo/Swift/CustomViewControllers/QueryCollectionViewController/SimpleCollectionViewController.swift]: `textLabel` here is an implicitly unwrapped optional - so no need for `?`

File: ParseUIDemo/Swift/CustomViewControllers/QueryCollectionViewController/StoryboardCollectionViewController.swift (hunk 11 above)
Comment (nlutsenko-fb) [inline ParseUIDemo/Swift/CustomViewControllers/QueryCollectionViewController/StoryboardCollectionViewController.swift]: `textLabel` here is an implicitly unwrapped optional - so no need for `?`

File: ParseUIDemo/Swift/CustomViewControllers/QueryCollectionViewController/StoryboardCollectionViewController.swift (hunk 12 above)
Comment (nlutsenko-fb) [inline ParseUIDemo/Swift/CustomViewControllers/QueryCollectionViewController/StoryboardCollectionViewController.swift]: `textLabel` here is an implicitly unwrapped optional - so no need for `?`

File: ParseUIDemo/Swift/CustomViewControllers/QueryCollectionViewController/StoryboardCollectionViewController.swift (hunk 13 above)
Comment (nlutsenko-fb) [inline ParseUIDemo/Swift/CustomViewControllers/QueryCollectionViewController/StoryboardCollectionViewController.swift]: `textLabel` here is an implicitly unwrapped optional - so no need for `?`

File: ParseUIDemo/Swift/CustomViewControllers/QueryCollectionViewController/SubtitleImageCollectionViewController.swift (hunk 14 above)
Comment (nlutsenko-fb) [inline ParseUIDemo/Swift/CustomViewControllers/QueryCollectionViewController/SubtitleImageCollectionViewController.swift]: `textLabel` here is an implicitly unwrapped optional - so no need for `?`

File: ParseUIDemo/Swift/CustomViewControllers/QueryCollectionViewController/SubtitleImageCollectionViewController.swift (hunk 15 above)
Comment (nlutsenko-fb) [inline ParseUIDemo/Swift/CustomViewControllers/QueryCollectionViewController/SubtitleImageCollectionViewController.swift]: `textLabel` here is an implicitly unwrapped optional - so no need for `?`

File: ParseUIDemo/Swift/CustomViewControllers/QueryCollectionViewController/SubtitleImageCollectionViewController.swift (hunk 16 above)
Comment (nlutsenko-fb) [inline ParseUIDemo/Swift/CustomViewControllers/QueryCollectionViewController/SubtitleImageCollectionViewController.swift]: `textLabel` here is an implicitly unwrapped optional - so no need for `?`

**Comment by hallucinogen** [comment]
Looks good to me! Ship it

**Pull request merged**